File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/AccessorEntity.java
package io.quarkus.it.panache;

import jakarta.persistence.Entity;
import jakarta.persistence.Transient;

@Entity
public class AccessorEntity extends GenericEntity<Integer> {

    public String string;
    public char c;
    public boolean bool;
    public byte b;
    public short s;
    public int i;
    public long l;
    public float f;
    public double d;
    @Transient
    public Object trans;
    @Transient
    public Object trans2;

    // FIXME: those appear to be mapped by hibernate
    transient int getBCalls = 0;
    transient int setICalls = 0;
    transient int getTransCalls = 0;
    transient int setTransCalls = 0;

    public void method() {
        // touch some fields
        @SuppressWarnings("unused")
        byte b2 = b;
        i = 2;
        t = 1;
        t2 = 2;
    }

    // explicit getter or setter

    public byte getB() {
        getBCalls++;
        return b;
    }

    public void setI(int i) {
        setICalls++;
        this.i = i;
    }

    public Object getTrans() {
        getTransCalls++;
        return trans;
    }

    public void setTrans(Object trans) {
        setTransCalls++;
        this.trans = trans;
    }
}

File: integration-tests/hibernate-orm-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/kotlin/AccessorEntity.kt
package io.quarkus.it.panache.kotlin

import jakarta.persistence.Entity
import jakarta.persistence.Transient

@Entity
open class AccessorEntity : GenericEntity<Int>() {
    var string: String? = null
    var c = 0.toChar()
    var bool = false
    var b: Byte = 0
        get() {
            getBCalls++
            return field
        }
    var s: Short = 0
    var i: Int = 0
        set(value) {
            setICalls++
            field = value
        }
    var l: Long
        get() {
            throw UnsupportedOperationException("just checking")
        }
        set(value) {
            throw UnsupportedOperationException("just checking")
        }
    var f = 0f
    var d = 0.0

    @Transient
    var trans: Any? = null

    @Transient
    var trans2: Any? = null

    // FIXME: those appear to be mapped by hibernate
    @Transient
    var getBCalls = 0

    @Transient
    var setICalls = 0

    @Transient
    var getTransCalls = 0

    @Transient
    var setTransCalls = 0

    fun method() { // touch some fields
        val b2 = b
        i = 2
        t = 1
        t2 = 2
    }
}
